numpyro/distributions/constraints.py

```diff
@@ -569,8 +600,8 @@ def __repr__(self) -> str:
         return fmt_string
 
 
-class _LowerCholesky(_SingletonConstraint):
-    event_dim = 2
+class _LowerCholesky(_SingletonConstraint[NonScalarArray]):
+    _event_dim = 2
```

Comment:
just curious: is there typing error if we do `event_dim=2` here? Using `event_dim=2` is more explicit I think